util.go

```diff
@@ -79,7 +88,14 @@ func isStatus(statusCode int, responseBody string) (bool, error) {
 	}
 
 	if responsePayload.StatusCode == statusCode {
-		return true, nil
+		if reason == "" {
+			// We're not looking for a specific reason, so we're done
+			return true, nil
+		}
+		// Match end of status text to ": " + <reason>
+		if strings.HasSuffix(responsePayload.StatusText, ": "+reason) {
```

Comment:
I've wanted to do that, but there is no requirement for either text or reason NOT to contain a colon.

That means that there can be many parts and you do not know which to compare.